tests/order_modifiers_integration/Fees/Create_Fees_Modifiers_Test.php

```diff
@@ -39,4 +43,66 @@ public function does_edit_screen_render_with_no_data() {
 		$snapshot = $this->does_edit_form_display_properly_with_no_data();
 		$this->assertMatchesHtmlSnapshot( $snapshot );
 	}
+
+	/**
+	 * @test
+	 * @return void
+	 */
+	public function does_table_render_correctly_with_search_and_page() {
+		$this->assertMatchesHtmlSnapshot( $this->get_table_display_with_search() );
+	}
+
+	protected function get_table_display_with_search(): string {
+		$ids = [];
+		for ( $i = 0; $i < 11; $i++ ) {
+			$data = [
+				'modifier'                    => $this->modifier_type,
+				'order_modifier_slug'         => sprintf( 'test_%1$s_%2$02d', $this->modifier_type, $i ),
+				'order_modifier_display_name' => uniqid( 'fee' ),
+			];
+
+			$applied_to = $i % 2 ? 'all' : 'per';
+
+			$ids[] = $this->upsert_order_modifier_for_test( $data, $applied_to )->id;
+		}
+
+		// Insert another modifier with a unique display name.
+		$ids[] = $this->upsert_order_modifier_for_test( [
+			'modifier'                    => $this->modifier_type,
+			'order_modifier_slug'         => 'test_fee_unique',
+			'order_modifier_display_name' => 'XXXX Unique Fee',
+		] )->id;
+
+		unset( $_REQUEST['s'], $_REQUEST['paged'], $_REQUEST['id'], $_REQUEST['edit'], $_POST['edit'] );
+
+		$_REQUEST = [
+			'modifier' => $this->modifier_type,
+			's'        => 'Unique Fee',
+			'paged'    => 3,
+		];
+
+		// We expect wp_redirect to be called.
+		$test  = $this;
+		$unset = $this->set_fn_return(
+			'wp_redirect',
+			function( $location ) use ( $test ) {
+				$test->assertEquals( '?paged=1', $location );
+				return true;
+			},
+			true
+		);
```

Comment:
This assertion does not tell us if the wp_redirect funciton was actually called. Localize a variable inside the callback by referrence and then after it the function should have been called make the assertion.

e.g.

```php
$locaiton = null;
$this->set_fn_return(
			'wp_redirect',
			function( $redirect_to ) use ( $location ) {
				$location = $redirect_to;
				return true;
			},
			true
		);
		
		
.... execute what should call wp_redirect here...

$this->assert( $location ); // now we are sure what we expect was called!
```